JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/GroupBean.java

```diff
@@ -8,132 +11,119 @@
 import java.util.HashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
-import ru.javafiddle.jpa.entity.Access;
-import ru.javafiddle.jpa.entity.UserGroup;
-import ru.javafiddle.jpa.entity.Group;
-import ru.javafiddle.jpa.entity.User;
+import ru.javafiddle.jpa.entity.*;
 
 
 @Stateless
 
 public class GroupBean {
 
-    @PersistenceContext
+    private static final Logger logger =
+            Logger.getLogger(ProjectBean.class.getName());
+    @PersistenceContext(name = "JFPersistenceUnit")
     EntityManager em;
 
-    public GroupBean() {
-    }
+    public GroupBean(){}
+
+//we always know that the group were we are trying to add a user exits now
+
+//CALL THIS METHOD WHEN THE PERSON IS CREATING A GROUP BY HIMSELF!!
 
-    //we should get at first userId, then add this element to UserGroups
-    //when we are in this method we definitly have group with such id
-    public void addMember(int groupId, String userNickName, String accessRights) {
+    @TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
+    public void addMember(int groupId, String groupName,String userNickName, String accessRights) {
 
-        //check if this group exists
-       /* Group group;
-        if (groupId == 0) {
-            group = new Group();
-            group.setGroupName("default");//still it is not clear about group names
-            em.persist(group);
-        }*/
+
+        Group group = getGroup(groupId);
+        if (group.getGroupId() == -1) {
+            createGroup(groupName);
+        }
         User user = getUser(userNickName);
         Access access = getAccess(accessRights);
 
+        UserGroup ug = new UserGroup(group, user, access);
 
-        Group g = getGroup(groupId);
-        UserGroup userGroup = (UserGroup)em.createQuery("SELECT us FROM UserGroup us WHERE us.group.groupId=:groupId");
-        /*Group group = getGroup(groupId);
-        if (group == null) {
-            group = new Group();
-            group.setGroupId(groupId);
-            group.setGroupName("default");//still it is not clear about group names
-        }*/
-        userGroup.setAccess(access);
 
-        em.getTransaction().begin();
-        em.persist(userGroup);
-        em.getTransaction().commit();
+        ug.setUserId(user.getUserId());
+        ug.setGroupId(group.getGroupId());
+
+        //  em.getTransaction().begin();
+        em.persist(ug);
 
-        //check if this group exists in list of groups
-        //group = (Group)em.createQuery("SELECT g from Group g WHERE g.groupId =:groupid")
-        //      .setParameter("groupid", groupId)
-        //      .getSingleResult();
-        //How can we get groupId from our services?
+        ug = getUserGroup(user.getUserId(),group.getGroupId());
+        addUserGroupToGroupList(group, ug);
 
 
     }
 
-    public  Map<String, String> getAllMembers(String groupId, String userNickName, String accessRights){
+    public  Map<String, String> getAllMembers(int groupId){
```

Comment:
Hm, the method name badly suits its purpose. I would name it ~getMemberAccessMap()